### PR TITLE
Add typehints

### PR DESCRIPTION
**Pull request by szepeviktor**
...and fix remove_action parameters

### CODE REVIEW

**Comment by szepeviktor** [comment]
These were also discovered by @phpstan

File: headache.php [hunk 1]
```diff
@@ -93,9 +93,14 @@ function headache_disable_feeds(): void
...
 remove_action('wp_head', 'wp_oembed_add_host_js');
 
-// Disable default users API endpoints for security.
-// https://www.wp-tweaks.com/hackers-can-find-your-wordpress-username/
-function headache_disable_rest_endpoints($endpoints): array
+/**
+ * Disable default users API endpoints for security.
+ *
+ * @link https://www.wp-tweaks.com/hackers-can-find-your-wordpress-username/
+ * @param array<string, array<callable, string>|list<array<callable, string>>> $endpoints
```
Comment (vinkla) [inline headache.php:100]: Please remove this line.

File: headache.php [hunk 2]
```diff
@@ -93,9 +93,14 @@ function headache_disable_feeds(): void
...
+ * @return array<string, array<callable, string>|list<array<callable, string>>>
```
Comment (vinkla) [inline headache.php:101]: Please remove this line.

**Comment by szepeviktor** [comment]
Remove.

I'm very sad that you are not (yet) a fan of static analysis.

**Comment by vinkla** [comment]
Thanks!

**Comment by szepeviktor** [comment]
You're welcome.

**Pull request merged**